src/pages/Graph/SummaryPanelNode.tsx

```diff
@@ -100,13 +102,31 @@ export class SummaryPanelNode extends React.Component<SummaryPanelNodeProps, Sum
             {shouldRenderService && <div>{renderBadgedLink(nodeData, NodeType.SERVICE)}</div>}
             {shouldRenderApp && <div>{renderBadgedLink(nodeData, NodeType.APP)}</div>}
             {shouldRenderWorkload && <div>{renderBadgedLink(nodeData, NodeType.WORKLOAD)}</div>}
+            {shouldRenderGatewayHostnames && this.renderGatewayHostnames(nodeData)}
+            {shouldRenderVsHostnames && this.renderVsHostnames(nodeData)}
           </div>
         </div>
         {shouldRenderTraces ? this.renderWithTabs(nodeData) : this.renderTrafficOnly()}
       </div>
     );
   }
 
+  private renderGatewayHostnames = (nodeData:  DecoratedGraphNodeData) => {
+    return (
+      <div>
+        {nodeData.isGateway?.ingressInfo?.hostnames?.map(hostname => renderBadgedHost(hostname))}
+      </div>
+    );
+  };
+
+  private renderVsHostnames = (nodeData:  DecoratedGraphNodeData) => {
+    return (
+      <div>
+        {nodeData.hasVS?.hostnames?.map(hostname => renderBadgedHost(hostname))}
+      </div>
+    );
+  };
+
```

Comment:
I think this where we need to be more creative to avoid a lot of vertical space and an annoying number of badges.   Maybe just one badge and if there is just one host, show it.  Otherwise show a fixed scrolling box with the entries?

Reply:
you mean using a comma separated list, rather than a vertical list with "bullets" (where the bullet is the badge)?

Reply:
We can do scrolling, but this is how it will look:

![image](https://user-images.githubusercontent.com/23639005/127232234-109b6166-d581-4863-a005-c1a1a5a7c3c0.png)

I don't like that much the double scrollbars in the side panel. But if it is OK for you, I can add them.

Reply:
What about an [expandable section](https://www.patternfly.org/v4/components/expandable-section)?  It could reduce the scrolling without using a secondary scroll.

Reply:
I think an expandable section could work, and would not need to scroll because, as you said, we already have vertical scrolling. The only suggestion I have is that we move it to the "badge summary area":

![image](https://user-images.githubusercontent.com/2104052/127353893-f462b865-7e9f-4ad1-8d4c-02b6ce03748b.png)

The above is just to show, as a reminder, that we have this section.  I think we should add an "Is Gateway" to this, and for gateways and relevant VS we could add an expandable section underneath.  > X hosts... or something like that.

What do you think?  Hopefully not a CSS nightmare.

Reply:
It shouldn't be hard. I'll try it.

Reply:
This is how it looks:

![image](https://user-images.githubusercontent.com/23639005/127366770-6c1223f2-6250-4757-beba-b89ee8e934a9.png)

It's in the last commit.

Reply:
Can we have it directly below the "Is gateway" and not detached with the Workload link in between?

Reply:
This is the updated look in commit https://github.com/kiali/kiali-ui/pull/2196/commits/01225bc02e5b8448123543fa9b66d4e85101f404

![image](https://user-images.githubusercontent.com/23639005/127374710-730a0355-c112-4194-856a-646c164cf2e2.png)

I also moved around the hosts list for the VirtualService/RequestRouting case.


> Any idea what happens if the value is long? Do we need ellipses and a tooltip?

The text should wrap to the next line... Probably it won't look that good.

Reply:
This looks good to me, let me when you are complete and I will pull and test.

Reply:
Done:

![image](https://user-images.githubusercontent.com/23639005/127385019-f552c6b4-063d-49bb-b76e-a38ebcfc8dba.png)